src/core/IO/Handle.pm6

```diff
@@ -116,6 +116,14 @@ my class IO::Handle {
           ),
         );
 
+        nqp::if(
+            nqp::if(:$truncate, :$create),
```

Comment:
`:$truncate` creates a Pair. That's not needed here. Also, this doesn't fix cases when only `$truncate` or only `$create` is specified.

You can also just use `&&` and `||` operators. They get converted to `nqp::if` and `nqp::unless` (the rest of this routine here does use too many `nqp::if`s because when I wrote that stuff I didn't know `&&` and `||` got converted to nqp ops)

So you can just write `($truncate || $create) && die ...` or `die ... if $truncate || $create` (that `if` would also get converted to `nqp::if`)

Reply:
To clarify, I think both `:truncate` and `:create`, even when used individually don't make much sense in read-only mode.

Reply:
I need to study the internals of nqm more so I can write optimized code the right way.

>To clarify, I think both :truncate and :create, even when used individually don't make much sense in read-only mode.

>  To clarify, I think both :truncate and :create, even when used individually don't make much sense in read-only mode.

I agree.

Reply:
> I need to study the internals of nqm more so I can write optimized code the right way.

You can dump QAST trees with `--target=optimize` and see what the stuff you write comes out as. It takes some time to get used to, but once you do, it's easy to see what gets optimized to what. There's an article on that: https://rakudo.party/post/Perl-6-Core-Hacking-QASTalicious